### DISCUSSION TITLE
Is there any function to detect speech and silence in Silero-Vad?

### DISCUSSION THREAD

**Discussion by harshraj3223** [snakers4/silero-vad#201] [Q&A] [answered]
Hey there, 
As you know, we use the method 'is_speech()' under 'webrtcvad' to detect if a specified interval is silent or has speech. Do we have anything similar to 'is_speech()' in 'Silero-vad' as well? I mean something that returns a bool value for a provided chunk? Please provide the method name, if available, or its suitable alternative.

**Comment by snakers4** [comment, marked as answer]
Hi, 

You can just rewrite this function to return a single bool value if the speech was detected:

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L119-L130

If you do not want to mess with this function, you can just write a wrapper that provides a True value if there is more speech than the specified threshold.

Also note that the VAD itself returns a bool value for a given audio chunk:

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L210

**Reply by harshraj3223** [reply]
Thanks a lot for the help!
So, I'm going to use speech_prob = model(chunk, sampling_rate).item() to decide if a given chunk is voiced or silent. Please find your time to answer the following questions. It'd really be a great help!!

(I) Can you please tell me all the arguments or parameters for this 'model()' function except for 'chunk' and 'sampling_rate'?

(ii) What would be the most efficient value for the threshold if I'm taking an audio stream from the microphone?

(iii) In the model(chunk, 16000) function, is it necessary to have the chunk value lie only in {512, 1024, 1536} ? If yes, what can be done if I want the model to wait for at least 1 second of silence/pause before turning the speech into a new line? (Context: I'm working on a speech-to-text model)

(iv) When I'm using Silero-VAD in my speech-to-text model, it doesn't seem to detect single words like 'Hey' or 'to', which are said in a short time interval. Also, when speaking a whole hefty sentence, it often fails to detect the beginning word of the sentence. Can you please guide me through this?

**Comment by snakers4** [comment]
> (I) Can you please tell me all the arguments or parameters for this 'model()' function except for 'chunk' and 'sampling_rate'?

The necessary parameters are listed in the above function.

> (ii) What would be the most efficient value for the threshold if I'm taking an audio stream from the microphone?

Depends on the application.
It can be easily tuned with the provided `visualize_probs` flag.

> (iii) In the model(chunk, 16000) function, is it necessary to have the chunk value lie only in {512, 1024, 1536} ?

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L153-L156

>  If yes, what can be done if I want the model to wait for at least 1 second of silence/pause before turning the speech into a new line? (Context: I'm working on a speech-to-text model)

Do not pass any chunks or start a new VAD session.

> (iv) When I'm using Silero-VAD in my speech-to-text model, it doesn't seem to detect single words like 'Hey' or 'to', which are said in a short time interval. Also, when speaking a whole hefty sentence, it often fails to detect the beginning word of the sentence. Can you please guide me through this?

Most likely you need to tune some of there parameters, see the `visualize_probs` flag:

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L121-L125
